### ISSUE TITLE
A11y_ Java Tooling for IntelliJ/Eclipse – New Feature_Spark Submission_Screen Reader: When focus is on ‘Artifact from Eclipse project’ radio button Narrator/NVDA is reading the label ‘Cluster Name’ along with it

### ISSUE THREAD

**Issue by Tanushagottipati** [microsoft/azure-tools-for-java#3558]
[Check out Accessibility Insights!](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Faccessibilityinsights.io%2F&data=02%7C01%7Cv-tagott%40microsoft.com%7Cb67b2c4b646d4f9561a208d6f4b5c39b%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636965458847261333&sdata=PJMiKjb64uwhBN98LZpctnBiAVeOvw%2BLBrnxBtrjl2M%3D&reserved=0) - Identify accessibility bugs before check-in and make bug fixing faster and easier

GitHubTags: #A11yTCS; #A11yMAS; #DesktopApp; #Win10; #A11y_JavaToolingforIntelliJ/Eclipse–NewFeature; #MAS1.3.1; #ScreenReader; #C+E-H; #JavatoolingforIntelliJ/Eclipse; #Narrator; #NVDA

**Environment Details:**
Eclipse IDE for Enterprise Java Developers.
Version: 2019-06 (4.12.0)
Build id: 20190614-1200
Solution Installed: The Azure Toolkit for Eclipse 3.17.0

**OS Details:** 
OS: Windows (Version 1903, Build: 18362.175)
Screen Reader: Narrator, NVDA
MAS Violated: MAS1.3.1

**Repro Steps:** 
1) Start Narrator/NVDA. Open Java EE - Eclipse application
2) Install 'Azure Toolkit for Eclipse' from Marketplace
3) Open Eclipse and install the Scala plug-in 
4) Tab till tools-->Azure-->Azure Sign in and sign into Azure account and select any subscription
5) In Azure Explorer section, tab till HDInsight and expand it. Select 'Spark on HDInsight (Scala)'. Tab till Next and press enter
6) Tab till Next button and press enter. Press Finish button
7) In Project Explorer, Expand the created Scala project right-click src, New-->Other and press enter
8) Tab till Scala Wizards, expand and select 'Scala Object' and press next
9) Enter the name of the file and press Finish button
10) Open Scala file and write any code in the editor
11) In Project Explorer view, right-click your project, choose Azure-->Submit Spark Application to HDInsight
12) Verify When focus is on ‘Artifact from Eclipse project’ radio button Narrator/NVDA is reading the label ‘Cluster Name’ along with it

**Actual Result:** 
When focus is on ‘Artifact from Eclipse project’ radio button Narrator/NVDA is reading the label ‘Cluster Name’ along with it

**Expected Result:** 
When focus is on ‘Artifact from Eclipse project’ radio button Narrator/NVDA should read the section name and label of the radio button but should not read the other labels present in the window

**User Impact:** 
Screen Reader users will find difficulty in understanding if screen reader is reading other controls labels when focus is on radio button

**MAS Reference:** 
https://microsoft.sharepoint.com/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={54f28d1f-a2d1-4dcd-84e1-5c9b87e8aba4}

[Bug_4_Narrator is reading the cluster name along with teh radio button label name.pptx](https://github.com/microsoft/azure-tools-for-java/files/3635520/Bug_4_Narrator.is.reading.the.cluster.name.along.with.teh.radio.button.label.name.pptx)

![Bug_4_NVDA is reading other label names](https://user-images.githubusercontent.com/41109135/65321495-fbbab880-dbc1-11e9-9fae-473c42d03c13.png)

**Comment by t-rufang** [comment]
The job view page is wrote as static web pages with the help of JavaFX WebView which has no support for Accessibility. Please refer to JDK issue [JDK-8177382](https://bugs.openjdk.java.net/browse/JDK-8177382)

**Comment by t-rufang** [comment]
Fixed in #3735, #3737 and  #3763

**Comment by Tanushagottipati** [comment]
Regressed in the below version and the issue is fixed. Hence closing the bug and attached latest screenshot for reference

Environment Details:
Eclipse IDE for Enterprise Java Developers.
Version: 2019-06 (4.12.0)
Build id: 20190614-1200
Solution Installed: The Azure Toolkit for Eclipse 3.19.0

![Screenshot (653)](https://user-images.githubusercontent.com/41109135/69145901-ab45d580-0af4-11ea-9722-12e4a47099c7.png)

**Comment by vscjavaci** [comment]
Track internal issue by AB#1749668
